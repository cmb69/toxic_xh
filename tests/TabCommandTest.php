<?php

namespace Toxic;

use ApprovalTests\Approvals;
use PHPUnit\Framework\TestCase;
use Plib\FakeRequest;
use Plib\View;

class TabCommandTest extends TestCase
{
    /** @var array<string,string> */
    private $conf;

    /** @var View */
    private $view;

    public function setUp(): void
    {
        $this->conf = XH_includeVar("./config/config.php", "plugin_cf")["toxic"];
        $plugin_tx = XH_includeVar("./languages/en.php", "plugin_tx");
        $this->view = new View("./views/", $plugin_tx["toxic"]);
    }

    private function sut(): TabCommand
    {
        $pageData = array('toxic_class' => 'test', "toxic_category" => "");
        return new TabCommand($this->conf, $pageData, $this->view);
    }

    public function testRendersClassInput(): void
    {
        $this->conf['classes_available'] = '';
        $request = new FakeRequest(["url" => "http://example.com/xh/?Welcome"]);
        Approvals::verifyHtml($this->sut()($request));
    }

    public function testRendersClassSelect(): void
    {
        $this->conf['classes_available'] = 'one,two,three,test';
        $request = new FakeRequest(["url" => "http://example.com/xh/?Welcome"]);
        Approvals::verifyHtml($this->sut()($request));
    }
}
