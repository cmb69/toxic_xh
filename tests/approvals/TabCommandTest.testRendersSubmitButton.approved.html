
<form id="toxic_tab" action="/xh/?Welcome" method="post">
  <p>
    <label>
      <span>Category</span>
      <input type="text" name="toxic_category" value="">
    </label>
  </p>
  <p>
    <label>
      <span>Class</span>
      <input type="text" name="toxic_class" value="test">
    </label>
  </p>
  <p class="toxic_tab_buttons">
    <button name="save_page_data">Save</button>
  </p>
</form>
